linker/linker_test.go

```diff
@@ -2234,6 +2234,579 @@ func TestLinkerValidation(t *testing.T) {
 			},
 			expectedErr: `test.proto:1:39: first value of open enum Foo must be zero`,
 		},
+		"success_extension_declarations": {
```

Comment:
Really nice test coverage. I can't think of something missing, and it all matches my understanding of extension declarations.